staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -83,3 +87,19 @@ func (record *attributesRecord) GetOldObject() runtime.Object {
 func (record *attributesRecord) GetUserInfo() user.Info {
 	return record.userInfo
 }
+
+func (record *attributesRecord) GetAnnotations() map[string]string {
+	return record.annotations
+}
+
+func (record *attributesRecord) SetAnnotations(plugin_name, key, value string) bool {
+	if record.annotations == nil {
+		return false
+	}
+	key = plugin_name + "/" + key
```

Comment:
plugin_name + ".admission.k8s.io/" + key ?

Reply:
We also have to check that the key is a DNS segment.

Reply:
> We also have to check that the key is a DNS segment.

Done.

> plugin_name + ".admission.k8s.io/" + key ?
`plugin_name` is already fully qualified. 
Webhook plugin do not use `.admission.k8s.io` in their plugin name, I think.